src/base/QXmppDataForm.cpp

```diff
@@ -637,8 +637,7 @@ void QXmppDataForm::toXml(QXmlStreamWriter *writer) const
             type == QXmppDataForm::Field::ListSingleField)
         {
             QPair<QString, QString> option;
-            foreach (option, field.options())
-            {
+            for (const auto &option : field.options()) {
```

Comment:
Note this shadows the other `option`. I guess the previous line can (and should) be removed.